entrypoint/cmd/entry.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"text/template"
+
+	ee "github.com/coreos/entrypoint/exec"
+	rhjobspec "github.com/coreos/entrypoint/spec"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+const cosaContainerDir = "/usr/lib/coreos-assembler"
+
+var (
+	version = "devel"
+
+	// Location of COSA
+	cosaDir string
+
+	// spec is an RHCOS spec
+	spec     rhjobspec.JobSpec
+	specFile string
+
+	shellCmd = []string{"/bin/bash", "-x"}
+
+	cmdRoot = &cobra.Command{
+		Use:   "entry [command]",
+		Short: "COSA entrypoint",
+		Long: `Entrypoint for CoreOS Assemlber
+Wrapper for COSA commands and templates`,
+		PersistentPreRun: preRun,
+	}
+
+	cmdVersion = &cobra.Command{
+		Use:   "version",
+		Short: "Print the version number and exit.",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Printf("entry/%s version %s\n",
+				cmd.Root().Name(), version)
+		},
+	}
+
+	cmdSingle = &cobra.Command{
+		Use:   "run",
+		Short: "Run the commands and bail",
+		Args:  cobra.MinimumNArgs(1),
+		Run:   runSingle,
+	}
+
+	cmdSteps = &cobra.Command{
+		Use:   "run-steps",
+		Short: "Run Steps from [file]",
+		Args:  cobra.MinimumNArgs(1),
+		Run:   runSteps,
+	}
+)
+
+func init() {
+	// cosaDir is set during a complile setting.
+	if cosaDir == "" {
+		path, err := os.Getwd()
+		if err != nil {
```

Comment:
Hmm...`os.Getwd()` can only fail I think in situations where the current directory is deleted underneath the process...why are we trying to handle that?

Reply:
being pendantic

Reply:
> being pendantic

@darkmuggle To clarify do you mean that you're being pedantic with extra checking?

Reply:
Correct, I'm being pedantic with the extra check. It doesn't hurt anything. I definitely wasn't saying that Mr. Walters was being pedantic in the review (which is one way to read my poorly formed sentence fragment.)

Reply:
Thanks!  To clarify I did read it the first way initially.

But my higher level uncertanty isn't checking for errors - that's always good IMO (now that I look I phrased this badly) so I more meant why are we silently falling back instead of propagating the error?   A whole lot of tools out there will fail if their current working directory is deleted - usually this is a user error.

IOW why not just:

```
if cosaDir == {
    path, err := os.Getwd()
    if err != nil {
        return err
    }
```

Not a big deal obviously, just one of those detail things that can be good to have a rough consensus on since the pattern tends to reoccur elsewhere.